packages/asset-buyer/src/swap_quoter.ts

```diff
@@ -0,0 +1,436 @@
+        return (await this._getSwapQuoteAsync(
+            makerAssetData,
+            takerAssetData,
+            takerAssetSellAmount,
+            'marketSell',
+            options,
+        )) as MarketSellSwapQuote;
+    }
+
+    /**
+     * Get a `SwapQuote` containing all information relevant to fulfilling a swap between a desired ERC20 token address and ERC20 owned by a provided address.
+     * You can then pass the `SwapQuote` to a `SwapQuoteConsumer` to execute a buy, or process SwapQuote for on-chain consumption.
+     * @param   makerAssetData           The makerAssetData of the desired asset to swap for (for more info: https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md).
+     * @param   takerAssetData           The takerAssetData of the asset to swap makerAssetData for (for more info: https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md).
+     * @param   makerAssetSwapAmount     The amount of maker asset to swap for.
+     * @param   options                  Options for the request. See type definition for more information.
+     *
+     * @return  An object that conforms to SwapQuote that satisfies the request. See type definition for more information.
+     */
+    public async getMarketBuySwapQuoteAsync(
+        makerAssetData: string,
+        takerAssetData: string,
+        makerAssetBuyAmount: BigNumber,
+        options: Partial<SwapQuoteRequestOpts> = {},
+    ): Promise<MarketBuySwapQuote> {
+        assert.isBigNumber('makerAssetBuyAmount', makerAssetBuyAmount);
+        return (await this._getSwapQuoteAsync(
+            makerAssetData,
+            takerAssetData,
+            makerAssetBuyAmount,
+            'marketBuy',
+            options,
+        )) as MarketBuySwapQuote;
+    }
+    /**
+     * Get a `SwapQuote` containing all information relevant to fulfilling a swap between a desired ERC20 token address and ERC20 owned by a provided address.
+     * You can then pass the `SwapQuote` to a `SwapQuoteConsumer` to execute a buy, or process SwapQuote for on-chain consumption.
+     * @param   makerAssetData           The makerAssetData of the desired asset to swap for (for more info: https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md).
+     * @param   takerAssetData           The takerAssetData of the asset to swap makerAssetData for (for more info: https://github.com/0xProject/0x-protocol-specification/blob/master/v2/v2-specification.md).
+     * @param   makerAssetSwapAmount     The amount of maker asset to swap for.
+     * @param   options                  Options for the request. See type definition for more information.
+     *
+     * @return  An object that conforms to SwapQuote that satisfies the request. See type definition for more information.
+     */
+    public async getMarketBuySwapQuoteForERC20TokenAddressAsync(
```

Comment:
do you think these are simpler functions to use over the developer encoding and using `getMarketBuySwapQuote`? might be worth considering the inverse of descriptive function names so this one is easier to use